pkg/clients/v2/cache.go

```diff
@@ -0,0 +1,383 @@
+		c.createStorageGatewayClient(c.clients[role][region].awsConfig),
+	)
+	return tagging.NewLimitedConcurrencyClient(c.clients[role][region].tagging, concurrencyLimit)
+}
+
+func (c *clientCache) GetAccountClient(region string, role config.Role) account.Client {
+	if !c.refreshed {
+		// if we have not refreshed then we need to lock in case we are accessing concurrently
+		c.mu.Lock()
+		defer c.mu.Unlock()
+	}
+	if client := c.clients[role][region].account; client != nil {
+		return client
+	}
+	c.clients[role][region].account = account_v2.NewClient(c.logger, c.createStsClient(c.clients[role][region].awsConfig))
+	return c.clients[role][region].account
+}
+
+func (c *clientCache) Refresh() {
+	if c.refreshed {
+		return
+	}
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	// Avoid double refresh in the event Refresh() is called concurrently
+	if c.refreshed {
+		return
+	}
+
+	for _, regionClients := range c.clients {
+		for _, cache := range regionClients {
+			cache.cloudwatch = cloudwatch_v2.NewClient(c.logger, c.createCloudwatchClient(cache.awsConfig))
+			if cache.onlyStatic {
+				continue
+			}
+
+			cache.tagging = tagging_v2.NewClient(
+				c.logger,
+				c.createTaggingClient(cache.awsConfig),
+				c.createAutoScalingClient(cache.awsConfig),
+				c.createAPIGatewayClient(cache.awsConfig),
+				c.createEC2Client(cache.awsConfig),
+				c.createDMSClient(cache.awsConfig),
+				c.createPrometheusClient(cache.awsConfig),
+				c.createStorageGatewayClient(cache.awsConfig),
+			)
+
+			cache.account = account_v2.NewClient(c.logger, c.createStsClient(cache.awsConfig))
+		}
+	}
+
+	c.refreshed = true
+	c.cleared = false
+}
+
+func (c *clientCache) Clear() {
+	if c.cleared {
+		return
+	}
+	// Prevent concurrent reads/write if clear is called during execution
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	// Avoid double clear in the event Refresh() is called concurrently
+	if c.cleared {
+		return
+	}
+
+	for _, regions := range c.clients {
+		for _, cache := range regions {
+			cache.cloudwatch = nil
+			cache.account = nil
+			cache.tagging = nil
+		}
+	}
+
+	c.refreshed = false
+	c.cleared = true
+}
+
+func (c *clientCache) createCloudwatchClient(regionConfig *aws.Config) *cloudwatch.Client {
+	return cloudwatch.NewFromConfig(*regionConfig, func(options *cloudwatch.Options) {
+		if c.logger.IsDebugEnabled() {
+			options.ClientLogMode = aws.LogRequestWithBody | aws.LogResponseWithBody
+		}
+		options.Retryer = retry.NewStandard(func(options *retry.StandardOptions) {
+			options.MaxAttempts = 5
+			options.MaxBackoff = 3 * time.Second
+
+			// existing settings
+			// TODO how to tell the difference between throttle and non-throttle errors now?
+			//	NumMaxRetries: 5
+			//	//MaxThrottleDelay and MinThrottleDelay used for throttle errors
+			//	MaxThrottleDelay: 10 * time.Second
+			//	MinThrottleDelay: 1 * time.Second
+			//	// For other errors
+			//	MaxRetryDelay: 3 * time.Second
+			//	MinRetryDelay: 1 * time.Second
```

Comment:
I kept this here for now because I still don't quite know how to port the existing config. I'm not sure it matters all that much though.